File: japiwznm/WznmQBlkAItem.java
/**
  * \file WznmQBlkAItem.h
  * Java API code for record of table TblWznmQBlkAItem
  * \author Alexander Wirthmueller
  * \date created: 27 Aug 2020
  * \date modified: 27 Aug 2020
  */

package apiwznm;

import org.w3c.dom.*;
import sbecore.*;

public class WznmQBlkAItem {
	
	public WznmQBlkAItem(
				int jnum
				, String srefIxVBasetype
				, String titIxVBasetype
				, String sref
				, String srefIxWznmVVartype
				, String titIxWznmVVartype
				, String stubRefWznmMControl
				, String stubRefWznmMVector
				, String stubRefWznmMFeed
				, String stubRefWznmMTable
				, String stubRefWznmMBlock
				, String Defval
				, String stubRefWznmMVectoritem
				, String Comment
			) {
		this.jnum = jnum;
		this.srefIxVBasetype = srefIxVBasetype;
		this.titIxVBasetype = titIxVBasetype;
		this.sref = sref;
		this.srefIxWznmVVartype = srefIxWznmVVartype;
		this.titIxWznmVVartype = titIxWznmVVartype;
		this.stubRefWznmMControl = stubRefWznmMControl;
		this.stubRefWznmMVector = stubRefWznmMVector;
		this.stubRefWznmMFeed = stubRefWznmMFeed;
		this.stubRefWznmMTable = stubRefWznmMTable;
		this.stubRefWznmMBlock = stubRefWznmMBlock;
		this.Defval = Defval;
		this.stubRefWznmMVectoritem = stubRefWznmMVectoritem;
		this.Comment = Comment;
	};

	public int jnum;
	public String srefIxVBasetype;
	public String titIxVBasetype;
	public String sref;
	public String srefIxWznmVVartype;
	public String titIxWznmVVartype;
	public String stubRefWznmMControl;
	public String stubRefWznmMVector;
	public String stubRefWznmMFeed;
	public String stubRefWznmMTable;
	public String stubRefWznmMBlock;
	public String Defval;
	public String stubRefWznmMVectoritem;
	public String Comment;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WznmQBlkAItem");

		if (Xmlio.checkXPath(doc, basexpath)) {
			srefIxVBasetype = Xmlio.extractStringUclc(doc, basexpath, "srefIxVBasetype", "typ", null, 0);
			titIxVBasetype = Xmlio.extractStringUclc(doc, basexpath, "titIxVBasetype", "typ2", null, 0);
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			srefIxWznmVVartype = Xmlio.extractStringUclc(doc, basexpath, "srefIxWznmVVartype", "vty", null, 0);
			titIxWznmVVartype = Xmlio.extractStringUclc(doc, basexpath, "titIxWznmVVartype", "vty2", null, 0);
			stubRefWznmMControl = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMControl", "con", null, 0);
			stubRefWznmMVector = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMVector", "vec", null, 0);
			stubRefWznmMFeed = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMFeed", "fed", null, 0);
			stubRefWznmMTable = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMTable", "tbl", null, 0);
			stubRefWznmMBlock = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMBlock", "bl2", null, 0);
			Defval = Xmlio.extractStringUclc(doc, basexpath, "Defval", "dfv", null, 0);
			stubRefWznmMVectoritem = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMVectoritem", "vit", null, 0);
			Comment = Xmlio.extractStringUclc(doc, basexpath, "Comment", "cmt", null, 0);
			
			return true;
		};

		return false;
	};
};

File: apiwznm/WznmQBlkAItem.cpp
/**
	* \file WznmQBlkAItem.cpp
	* API code for table TblWznmQBlkAItem (implementation)
	* \copyright (C) 2016-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

#include "WznmQBlkAItem.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WznmQBlkAItem
 ******************************************************************************/

WznmQBlkAItem::WznmQBlkAItem(
			const uint jnum
			, const string srefIxVBasetype
			, const string titIxVBasetype
			, const string sref
			, const string srefIxWznmVVartype
			, const string titIxWznmVVartype
			, const string stubRefWznmMControl
			, const string stubRefWznmMVector
			, const string stubRefWznmMFeed
			, const string stubRefWznmMTable
			, const string stubRefWznmMBlock
			, const string Defval
			, const string stubRefWznmMVectoritem
			, const string Comment
		) {
	this->jnum = jnum;
	this->srefIxVBasetype = srefIxVBasetype;
	this->titIxVBasetype = titIxVBasetype;
	this->sref = sref;
	this->srefIxWznmVVartype = srefIxWznmVVartype;
	this->titIxWznmVVartype = titIxWznmVVartype;
	this->stubRefWznmMControl = stubRefWznmMControl;
	this->stubRefWznmMVector = stubRefWznmMVector;
	this->stubRefWznmMFeed = stubRefWznmMFeed;
	this->stubRefWznmMTable = stubRefWznmMTable;
	this->stubRefWznmMBlock = stubRefWznmMBlock;
	this->Defval = Defval;
	this->stubRefWznmMVectoritem = stubRefWznmMVectoritem;
	this->Comment = Comment;
};

bool WznmQBlkAItem::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WznmQBlkAItem");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "srefIxVBasetype", "typ", srefIxVBasetype);
		extractStringUclc(docctx, basexpath, "titIxVBasetype", "typ2", titIxVBasetype);
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "srefIxWznmVVartype", "vty", srefIxWznmVVartype);
		extractStringUclc(docctx, basexpath, "titIxWznmVVartype", "vty2", titIxWznmVVartype);
		extractStringUclc(docctx, basexpath, "stubRefWznmMControl", "con", stubRefWznmMControl);
		extractStringUclc(docctx, basexpath, "stubRefWznmMVector", "vec", stubRefWznmMVector);
		extractStringUclc(docctx, basexpath, "stubRefWznmMFeed", "fed", stubRefWznmMFeed);
		extractStringUclc(docctx, basexpath, "stubRefWznmMTable", "tbl", stubRefWznmMTable);
		extractStringUclc(docctx, basexpath, "stubRefWznmMBlock", "bl2", stubRefWznmMBlock);
		extractStringUclc(docctx, basexpath, "Defval", "dfv", Defval);
		extractStringUclc(docctx, basexpath, "stubRefWznmMVectoritem", "vit", stubRefWznmMVectoritem);
		extractStringUclc(docctx, basexpath, "Comment", "cmt", Comment);
	};

	return basefound;
};

/******************************************************************************
 class ListWznmQBlkAItem
 ******************************************************************************/

ListWznmQBlkAItem::ListWznmQBlkAItem() {
};

ListWznmQBlkAItem::ListWznmQBlkAItem(
			const ListWznmQBlkAItem& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQBlkAItem(*(src.nodes[i]));
};

ListWznmQBlkAItem& ListWznmQBlkAItem::operator=(
			const ListWznmQBlkAItem& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQBlkAItem(*(src.nodes[i]));

	return *this;
};

ListWznmQBlkAItem::~ListWznmQBlkAItem() {
	clear();
};

void ListWznmQBlkAItem::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWznmQBlkAItem::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WznmQBlkAItem* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWznmQBlkAItem");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WznmQBlkAItem", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WznmQBlkAItem(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WznmQBlkAItem[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};
